docs/4.0/components/forms.md

```diff
@@ -700,7 +700,7 @@ While Bootstrap will apply these styles in all browsers, Internet Explorer 11 an
 Provide valuable, actionable feedback to your users with HTML5 form validation–[available in all our supported browsers](https://caniuse.com/#feat=form-validation). Choose from the browser default validation feedback, or implement custom messages with our built-in classes and starter JavaScript.
 
 {% callout warning %}
-We **highly recommend** custom validation styles as native browser defaults are not announced to screen readers.
+We **highly recommend** custom validation styles, as native browser defaults are not announced to screen readers. Please ensure you manage the `aria-invalid` and `aria-describedby` attribute on `input` element according to the validation state.
```

Comment:
may actually be good to soften this warning a bit, as some browsers now seem to work ok. suggest `We recommend using custom validation styles, as native browser validation errors are not consistently announced by all browser and screen reader combinations.`

Reply:
yes, will finish to test that. Actually doesn't work properly with FFESR52 and NVDA 2017 even with the aria-invalid false, the browser still announce field as invalid when the user first get in the input field

Reply:
Wording updated. But also a bug filled to Mozilla. https://bugzilla.mozilla.org/show_bug.cgi?id=1413193
By my tests, aria-invalid status works well on Chrome and IE11, but don't on Firefox with NVDA or Jaws